portfolio/src/main/webapp/script.js

```diff
@@ -16,21 +16,70 @@
  * Adds a random fact to the page.
  */
 function addRandomFact() {
-  const facts =
-    [
-        'I am Romanian.', 
-        'Guitar and piano are some instruments that I play.', 
-        'I am really tall (180 cm).', 
-        'Friends call me by my middle name.',
-        'I have a dog called Zorro.',
-        'I am a STEP intern at Google.',
-        'I have two sisters, Sofia and Miruna.'
-    ];
-
-  // Pick a random fact.
-  const fact = facts[Math.floor(Math.random() * facts.length)];
-
-  // Add it to the page.
-  const factContainer = document.getElementById('fact-container');
-  factContainer.innerText = fact;
+    const facts =
+        [
+            'I am Romanian.', 
+            'Guitar and piano are some instruments that I play.', 
+            'I am really tall (180 cm).', 
+            'Friends call me by my middle name.',
+            'I have a dog called Zorro.',
+            'I am a STEP intern at Google.',
+            'I have two sisters, Sofia and Miruna.'
+        ];
+
+    // Pick a random fact.
+    const fact = facts[Math.floor(Math.random() * facts.length)];
+
+    // Add it to the page.
+    const factContainer = document.getElementById('fact-container');
+    factContainer.innerText = fact;
+}
+
+var slideIndex = 0;
+var slides = document.getElementsByClassName("mySlides");
+
+function showSlides() {
+    if(hasSlides()) {
+        hideAllSlides();
+        checkLastImage();
+        displayOneSlide();
+        goToNextSlide();
+        setTimeout(showSlides, 3000); // Change image every 3 seconds
+    } else {
+        alert("There are no images!");
+    }
+}
+
+function hideAllSlides() {
+    for (var i = 0; i < slides.length; i++) {
+        slides[i].style.display = "none";  
+    }
+}
+
+function checkLastImage() {
+    if (slideIndex > slides.length - 1) {
+        slideIndex = 0
+  }
 }
+
+function displayOneSlide() {
+    slides[slideIndex].style.display = "block";
+}
+
+function goToNextSlide() {
+    slideIndex++;
+}
+
+function hasSlides() {
+    if(slides === undefined || slides.length == 0) {
+        return false;
+    }
+
+    return true;
+}
+
+function getGreeting() {
+  fetch("/data").then(response => response.text()).then((greeting) => {
```

Comment:
Nice, I learnt about arrow functions just today in a tutorial.

The JS style guide also says that

> Tip: Always using parentheses even for single-parameter arrow functions can avoid situations where adding parameters, but forgetting to add parentheses, may result in parseable code which no longer works as intended.

so that's what I learnt when looking through your code.